Functions/QuickScript/SteamScript/script.js

```diff
@@ -20,10 +21,59 @@ SteamScript.prototype.constructor = SteamScript;
 SteamScript.prototype.appId = function(appId) {
     this._appId = appId;
     this._applicationHomepage = "http://store.steampowered.com/app/" + appId;
-    this._executableArgs = "steam://rungameid/" + this._appId;
+    this._executableArgs = ["-silent", "steam://rungameid/" + this._appId];
     return this;
 };
 
+SteamScript.prototype.getBytesToDownload = function(wine) {
+    // wait until download started
+    while (!fileExists(wine.prefixDirectory + "/drive_c/" + wine.getProgramFiles() + "/Steam/steamapps/appmanifest_" + this._appId + ".acf"))
+    {
+    }
+
+    var processBuilder = new java.lang.ProcessBuilder(["bash", "-c", "cat appmanifest_" + this._appId + ".acf | grep BytesToDownload | egrep -o '[0-9]+'"]);
+
+    processBuilder.directory(new java.io.File(wine.prefixDirectory + "/drive_c/" + wine.getProgramFiles() + "/Steam/steamapps"));
+
+    // make sure that BytesToDownload is set
+    var bytesToDownload = 0;
+    while (bytesToDownload == 0)
+    {
+        var process = processBuilder.start();
+        bytesToDownload = Number(org.apache.commons.io.IOUtils.toString(process.getInputStream()).trim());
+    }
+    return bytesToDownload;
+};
+
+SteamScript.prototype.getBytesDownloaded = function(wine) {
+    var downloadFolder = wine.prefixDirectory + "/drive_c/" + wine.getProgramFiles() + "/Steam/steamapps/downloading/" + this._appId;
+    // download folder is not yet/no more available
+    if (!fileExists(downloadFolder))
+    {
+        // check if download already finished (download folder has been deleted)
+        if (fileExists(wine.prefixDirectory + "/drive_c/" + wine.getProgramFiles() + "/Steam/steamapps/appmanifest_" + this._appId + ".acf"))
+        {
+            var processBuilder = new java.lang.ProcessBuilder(["bash", "-c", "cat appmanifest_" + this._appId + ".acf | grep BytesDownloaded | egrep -o '[0-9]+'"]);
+
+            processBuilder.directory(new java.io.File(wine.prefixDirectory + "/drive_c/" + wine.getProgramFiles() + "/Steam/steamapps"));
+
+            var process = processBuilder.start();
+            return Number(org.apache.commons.io.IOUtils.toString(process.getInputStream()).trim());
+        }
+        else
+        {
+            return 0;
+        }
+    }
+
+    var processBuilder = new java.lang.ProcessBuilder(["bash", "-c", "du -sb *_Data | egrep -o '[0-9]+'"]);
```

Comment:
I was more thinking about Files.size(File directory)

DirectoryWatcherSize needs to needs to be updated but the way it works is that it opens a new Thread, regularly checks a directory file size and accepts a callback you have defined.

It is definitely a good long term solution but AbstractDirectoryWatcher seems to be broken. We also need to create the corresponding factories